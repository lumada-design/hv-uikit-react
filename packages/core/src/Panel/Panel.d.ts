import { BoxProps, StandardProps } from "@mui/material";

export type HvPanelClassKey = "root";

export type HvPanelProps = StandardProps<BoxProps, HvPanelClassKey>;

export default function HvPanel(props: HvPanelProps): JSX.Element | null;
