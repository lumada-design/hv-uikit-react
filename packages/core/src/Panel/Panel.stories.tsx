import styled from "@emotion/styled";
import { Close, Edit } from "@hitachivantara/uikit-react-icons";
import { Meta, StoryObj } from "@storybook/react";
import {
  HvButton,
  HvPanel,
  HvPanelProps,
  HvTypography,
  theme,
} from "@hitachivantara/uikit-react-core";

const meta: Meta<typeof HvPanel> = {
  title: "Components/Panel",
  component: HvPanel,
};
export default meta;

export const Main: StoryObj<HvPanelProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    return (
      <HvPanel>
        <HvTypography>Panel Content</HvTypography>
      </HvPanel>
    );
  },
};

export const WithScroll: StoryObj<HvPanelProps> = {
  render: () => {
    return (
      <HvPanel style={{ width: "400px", height: "400px" }}>
        <div style={{ height: 600, backgroundColor: theme.colors.atmo4 }}>
          &nbsp;
        </div>
      </HvPanel>
    );
  },
};

const StyledButton = styled(HvButton)({
  position: "absolute",
  top: theme.space.sm,
  right: theme.space.sm,
  width: "32px",
  height: "32px",
});

export const FullWidth: StoryObj<HvPanelProps> = {
  render: () => {
    return (
      <HvPanel style={{ width: "100%", height: "200px" }}>
        <HvTypography>Panel Content</HvTypography>
        <StyledButton icon aria-label="Edit" variant="secondaryGhost">
          <Edit />
        </StyledButton>
      </HvPanel>
    );
  },
};

const CloseButton = styled(HvButton)({
  position: "absolute",
  top: theme.space.sm,
  right: theme.space.sm,
  width: "32px",
  height: "32px",
});

const Overlay = styled("div")({
  backgroundColor: theme.colors.atmo3,
  opacity: 0.8,
  width: "100%",
  padding: theme.space.md,
});

export const Modal: StoryObj<HvPanelProps> = {
  render: () => {
    return (
      <Overlay>
        <HvPanel
          style={{
            width: "100%",
            height: "200px",
            boxShadow: theme.colors.shadow,
          }}
        >
          <HvTypography>Panel Content</HvTypography>
          <CloseButton icon aria-label="Close" variant="secondaryGhost">
            <Close />
          </CloseButton>
        </HvPanel>
      </Overlay>
    );
  },
};
