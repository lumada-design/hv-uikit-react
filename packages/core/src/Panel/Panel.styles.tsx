import { theme } from "@hitachivantara/uikit-styles";

import { createClasses } from "../utils/classes";

export const { useClasses, staticClasses } = createClasses("HvPanel", {
  root: {
    position: "relative",
    padding: theme.space.sm,
    backgroundColor: theme.colors.atmo1,
    overflow: "auto",
  },
});
