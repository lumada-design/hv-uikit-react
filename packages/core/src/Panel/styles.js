const styles = (theme) => ({
  root: {
    position: "relative",
    padding: theme.hv.spacing.sm,
    backgroundColor: theme.hv.palette.atmosphere.atmo1,
    overflow: "auto",
  },
});

export default styles;
