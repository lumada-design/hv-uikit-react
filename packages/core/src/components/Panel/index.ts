export { default as panelClasses } from "./panelClasses";
export * from "./panelClasses";
export * from "./Panel";
