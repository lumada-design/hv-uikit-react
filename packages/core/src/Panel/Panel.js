import React from "react";
import clsx from "clsx";
import PropTypes from "prop-types";
import { Box, withStyles } from "@material-ui/core";
import styles from "./styles";

/**
 * A panel is a container used in a variety of patterns (e.g. dropdown, filter group, details section).
 * It can be horizontal or vertical and its size is defined by the content it holds inside and how it relates with surrounding patterns.
 * Regardless of its content, a panel look and feel should be consistent.
 */
const HvPanel = (props) => {
  const { id, className, classes, children, ...others } = props;

  return (
    <Box id={id} className={clsx(className, classes.root)} {...others}>
      {children}
    </Box>
  );
};

HvPanel.propTypes = {
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * The content of the panel.
   */
  children: PropTypes.node,
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
  }).isRequired,
};

export default withStyles(styles, { name: "HvPanel" })(HvPanel);
