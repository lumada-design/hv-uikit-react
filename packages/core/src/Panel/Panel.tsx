import { useDefaultProps } from "@core/hooks/useDefaultProps";

import { HvBaseProps } from "@core/types/generic";

import { ExtractNames } from "@core/utils/classes";

import { staticClasses, useClasses } from "./Panel.styles";

export { staticClasses as panelClasses };

export type HvPanelClasses = ExtractNames<typeof useClasses>;

export interface HvPanelProps extends HvBaseProps {
  /** A Jss Object used to override or extend the styles applied. */
  classes?: HvPanelClasses;
}

/**
 * A panel is a container used in a variety of patterns (e.g. dropdown, filter group, details section).
 * It can be horizontal or vertical and its size is defined by its content and how it relates to surrounding patterns.
 * Regardless of its content, a panel look and feel should be consistent.
 */
export const HvPanel = (props: HvPanelProps) => {
  const {
    id,
    className,
    classes: classesProp,
    children,
    ...others
  } = useDefaultProps("HvPanel", props);
  const { classes, cx } = useClasses(classesProp);

  return (
    <div id={id} className={cx(classes.root, className)} {...others}>
      {children}
    </div>
  );
};
