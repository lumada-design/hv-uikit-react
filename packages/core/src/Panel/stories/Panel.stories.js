import React from "react";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import { Edit, Close } from "@hitachivantara/uikit-react-icons";
import { HvPanel, HvButton, HvTypography } from "../..";

export default {
  title: "Components/Panel",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvPanel } from "@hitachivantara/uikit-react-core"',
    maturityStatus: "stable",
    dsVersion: "3.2.0",
  },
  component: HvPanel,
};

export const Main = () => (
  <HvPanel>
    <HvTypography>Panel Content</HvTypography>
  </HvPanel>
);

export const WithScroll = () => {
  const theme = useTheme();

  return (
    <HvPanel width="400px" height="400px">
      <div style={{ height: 600, backgroundColor: theme.hv.palette.atmosphere.atmo4 }}>&nbsp;</div>
    </HvPanel>
  );
};

export const FullWidth = () => {
  const useStyles = makeStyles((theme) => ({
    editButton: {
      position: "absolute",
      top: theme.hv.spacing.sm,
      right: theme.hv.spacing.sm,
      width: "32px",
      height: "32px",
    },
  }));

  const classes = useStyles();

  return (
    <HvPanel width="100%" height="200px">
      <HvTypography>Panel Content</HvTypography>
      <HvButton icon className={classes.editButton} aria-label="Edit">
        <Edit />
      </HvButton>
    </HvPanel>
  );
};

export const Modal = () => {
  const useStyles = makeStyles((theme) => ({
    closeButton: {
      position: "absolute",
      top: theme.hv.spacing.sm,
      right: theme.hv.spacing.sm,
      width: "32px",
      height: "32px",
    },
    overlay: {
      backgroundColor: theme.hv.palette.atmosphere.atmo3,
      opacity: 0.8,
      width: "100%",
      padding: theme.hv.spacing.md,
    },
  }));

  const theme = useTheme();

  const classes = useStyles();

  return (
    <div className={classes.overlay}>
      <HvPanel width="100%" height="200px" boxShadow={theme.hv.shadows[1]}>
        <HvTypography>Panel Content</HvTypography>
        <HvButton icon className={classes.closeButton} aria-label="Close">
          <Close />
        </HvButton>
      </HvPanel>
    </div>
  );
};
