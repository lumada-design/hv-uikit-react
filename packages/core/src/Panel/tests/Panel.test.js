/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";
import { Main } from "../stories/Panel.stories";
import { HvProvider, HvPanel } from "../..";

describe("Panel", () => {
  let wrapper;

  it("should be defined", () => {
    wrapper = mount(
      <HvProvider disableCssBaseline>
        <Main />
      </HvProvider>
    );
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvPanel)).toMatchSnapshot();
  });

  it("should render the Panel", () => {
    const component = wrapper.find(HvPanel);
    expect(component.length).toBe(1);
  });
});
