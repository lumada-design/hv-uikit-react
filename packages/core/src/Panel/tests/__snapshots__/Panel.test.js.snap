// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render correctly 1`] = `
<WithStyles(HvPanel)>
  <HvPanel
    classes={
      Object {
        "root": "HvPanel-root",
      }
    }
  >
    <Styled(MuiBox)
      className="HvPanel-root"
    >
      <div
        className="MuiBox-root MuiBox-root-1 HvPanel-root"
      >
        Panel Content
      </div>
    </Styled(MuiBox)>
  </HvPanel>
</WithStyles(HvPanel)>
`;
